### ISSUE TITLE
Regression: linking Java sources broken in 1.7.0

### ISSUE THREAD

**Issue by DanySK** [Kotlin/dokka#2544]
**Describe the bug**

Dokka used to be able to link Java sources, see, e.g., https://alchemistsimulator.github.io/reference/kdoc/alchemist/it.unibo.alchemist.core.implementations/-engine/add-output-monitor.html

Since 1.7.0, `dokkaHtmlCollector` fails with `textRange must not be null`.

**Expected behaviour**

Dokka does not fail, the generated documentation links Java sources as it used to do in 1.6.x

**To Reproduce**

```
git clone https://github.com/AlchemistSimulator/Alchemist.git
cd Alchemist
git checkout renovate/dokka
./gradlew dokkaHtmlCollector
```

**Dokka configuration**
Configuration of dokka used to reproduce the bug

```Kotlin
    tasks.withType<org.jetbrains.dokka.gradle.DokkaTask>().configureEach {
        dokkaSourceSets.configureEach {
            jdkVersion.set(multiJvm.jvmVersionForCompilation)
            listOf("kotlin", "java") // remove "java" from this list to "fix" the bug!
                .map { "src/main/$it" }
                .map { it to File(projectDir, it) }
                .toMap()
                .filterValues { it.exists() }
                .forEach { (path, file) ->
                    sourceLink {
                        localDirectory.set(file)
                        val project = if (project == rootProject) "" else project.name
                        val url = "https://github.com/AlchemistSimulator/Alchemist/blob/master/$project/$path"
                        remoteUrl.set(uri(url).toURL())
                        remoteLineSuffix.set("#L")
                    }
                }
            configurations.implementation.get().dependencies.forEach { dep ->
                val javadocIOURLs = fetchJavadocIOForDependency(dep)
                if (javadocIOURLs != null) {
                    val (javadoc, packageList) = javadocIOURLs
                    externalDocumentationLink {
                        url.set(javadoc)
                        packageListUrl.set(packageList)
                    }
                }
            }
        }
        failOnWarning.set(true)
    }
```

**Note**: removing `"java"` from the list above "fixes" the issue (it just turns off the source linking for Java sources).

**Installation**
- see Gradle scan:  https://gradle.com/s/4easgt5kurdlw
- Dokka version: 1.7.0

**Additional context**

Mixed Java-Kotlin projects (originally in Java, new parts written in Kotlin, slowly transitating to Kt).

**Are you willing to provide a PR?**

I have no idea what to fix, I need guidance at the very least.

**Comment by IgnatBeresnev** [comment]
Hi! Thanks for the report (your project is rather good for testing releases on, I must admit), this one was not expected :) 

The change must've come from bumping `intellij` platform from 211 to 213, we use it to analyze Java sources among other things. [This commit to be exact](https://github.com/JetBrains/intellij-community/commit/9baa4828daacd63c0d90112a6bb49e8c467ac126#diff-7b6daeb2d1bb1ad234a78ddbf659ef7d2563d75be2a2b4418fd5c46c88202a3fL236)

`EnumSyntheticMethod` used to be `extends LightMethod`, which had overridden `getTextRange()`
```kotlin
@Override
public TextRange getTextRange() {
  return myMethod.getTextRange();
}
```

But I can see now that it's been changed to `extends LightElement, implements PsiMethod`, and `LightElement` only has the default implementation which returns null:
```kotlin
@Override
public TextRange getTextRange() {
  return null;
}
```

Thus NPE.

We should probably just ignore all synthetic elements. I'll fix it and it'll be released in 1.7.10 in roughly two weeks.

___

Debug screenshot for future references:
![2022-06-21_19-28-11](https://user-images.githubusercontent.com/22685910/174860793-7a61ed27-33fd-4cc2-9e0d-2f81150e5c21.png)

**Comment by DanySK** [comment]
@IgnatBeresnev nice bug hunt :gun: 

Yeah, if you provide me with a tool I usually stress it as much as I can :smile: 

I don't know how the test suite of Dokka is built, but, in general, it should be possible to import external projects as submodules, then, when testing with the Gradle toolkit, patch them to run the current in-development version of a Gradle plugin. I do something similar with my own plugins.

Issue-wise, I'll be waiting for 1.7.10 (to be fair, my update bot will, I will hopefully just observe the result :smile: )
